akka-stream-testkit/src/test/scala/akka/stream/testkit/StreamSpec.scala

```diff
@@ -16,13 +16,16 @@ import scala.concurrent.duration._
 
 class StreamSpec(_system: ActorSystem) extends AkkaSpec(_system) {
   def this(config: Config) =
-    this(ActorSystem(AkkaSpec.getCallerName(getClass), ConfigFactory.load(config.withFallback(AkkaSpec.testConf))))
+    this(
+      ActorSystem(
+        AkkaSpec.getCallerName(classOf[StreamSpec]),
+        ConfigFactory.load(config.withFallback(AkkaSpec.testConf))))
 
   def this(s: String) = this(ConfigFactory.parseString(s))
 
   def this(configMap: Map[String, _]) = this(AkkaSpec.mapToConfig(configMap))
 
-  def this() = this(ActorSystem(AkkaSpec.getCallerName(getClass), AkkaSpec.testConf))
+  def this() = this(ActorSystem(AkkaSpec.getCallerName(classOf[StreamSpec]), AkkaSpec.testConf))
```

Comment:
Hmm, this is a relatively big change in behavior though, as now each test will be run with an actorsystem named `StreamSpec` instead of named after the actual spec it's in. Weird that it worked before I guess though?

Reply:
The same pattern is used in `AkkaSpec` itself.

I can take a stab at refactoring to keep the behavior but postpone any `getClass` to the actual constructor

Reply:
Yes, sorry I didn't get a chance to exercise the tests, so I was guessing.

Reply:
No problem, thanks a lot for getting the ball rolling in the first place!

Reply:
Ah, dang I didn't think when I looked at it. Yes we need to do that, thanks for picking it up @raboof

Reply:
Before it was just Predef, but maybe this will explain some previous failure. This is just inspecting the stack, but there is more recent JDK API for that.

Reply:
You're right looking at this more closely it probably broke way earlier and apparently we just didn't notice.. anyway #28355 should do the trick.